tests/test_feature_operations.py

```diff
@@ -0,0 +1,84 @@
+import re
+
+import numpy as np
+import pandas as pd
+import pytest
+
+from pyam import IamDataFrame
+
+# list of todo's here:
+#   - what happens if indexes are different (NotImplementedError?)
+
+# list of todo's for future PRs:
+#   - subtracting Series
+#   - subtracting floats
+#   - subtracting vectors
+#   - subtracting arrays
+#   - other operations e.g. multiplication, addition, division
+
+def test_subtraction_with_self(check_aggregate_df):
```

Comment:
`check_aggregate_df` is a less-than-ideal test case, because it's too large to quickly grasp what a function should do on it. Any chance you could use this opportunity to reduce it to one page of data (and harmonise how the model-and-scenario columns are defined)?

https://github.com/IAMconsortium/pyam/blob/a4e266c93f531f94285311fd354aaa881aa403c4/tests/conftest.py#L42

Reply:
It's kind of the point of it. The aggregate functions have to deal with a whole heap of edge cases so building a test case for it is hard and requires a fairly large dataframe. I agree that it's not great for a unit test, but it's the only way to do an integration test on the aggregate functions as far as I can tell. I'm happy to move the aggregation feature tests into a `tests/integration` folder and use a different test dataframe for testing the operations. However I think the operations will have to deal with a lot of edge cases too so having some integration tests on them could be worthwhile too.